client/src/core/services/robiot.service.ts

```diff
@@ -0,0 +1,145 @@
+
+@Injectable({
+    providedIn: 'root',
+})
+export class RobiotService {
+    private readonly resourceUrl;
+
+    constructor(private http: HttpClient) {
+        this.resourceUrl = environment.apiUrl;
+    }
+
+    // Gets the value of abscissa from a config's content value
+    getXCoordFromConfig(config: Configuration): string {
+        // Checks that the provided config is form a position
+        if (config.id == 301) {
+            return config?.content.split(',')[0];
+        }
+    }
+
+    // Gets the value of ordinate from a config's content value
+    getYCoordFromConfig(config: Configuration): string {
+        // Checks that the provided config is form a position
+        if (config.id == 301) {
+            return config?.content.split(',')[1];
+        }
+    }
+
+    getConfig(id: number): Observable<Configuration> {
+        return this.http.get<Configuration>(this.resourceUrl + '/robiot/' + id);
+    }
+
+    saveConfig(config: Configuration) {
+        return this.http.put(this.resourceUrl + '/robiot/' + config.id, config);
+    }
+
+    // Creates a bi-dimensionnal array representing a terrain from a terrain object
+    buildArrayFromTerrain(terrain: Terrain): Array<Array<string>> {
+        const map = Array.from({ length: terrain.width }, () =>
+            Array.from({ length: terrain.height }, () => CaseValue.EMPTY),
+        );
+
+        // Start
+        map[0][0] = CaseValue.START;
+
+        // Obstacles
+        for (const obstacle of terrain.obstacles) {
+            switch (obstacle.direction) {
+                case Direction.DOWN: {
+                    for (let i = obstacle.startY; i < obstacle.startY + obstacle.length; i++) {
+                        map[obstacle.startX][i] = CaseValue.OBSTACLE;
+                    }
+                    break;
+                }
+                case Direction.UP: {
+                    for (let i = obstacle.startY; i > obstacle.startY - obstacle.length; i--) {
+                        map[obstacle.startX][i] = CaseValue.OBSTACLE;
+                    }
+                    break;
+                }
+                case Direction.LEFT: {
+                    for (let i = obstacle.startX; i > obstacle.startX - obstacle.length; i--) {
+                        map[i][obstacle.startY] = CaseValue.OBSTACLE;
+                    }
+                    break;
+                }
+                case Direction.RIGHT: {
+                    for (let i = obstacle.startX; i < obstacle.startX + obstacle.length; i++) {
+                        map[i][obstacle.startY] = CaseValue.OBSTACLE;
+                    }
+                    break;
+                }
+            }
+        }
+
+        // Trees
+        for (const tree of terrain.trees) {
+            map[tree.x][tree.y] = CaseValue.TREE;
+        }
+
+        return map;
+    }
+
+    // Checks if a destination is obtainable (not out of bound, doesnt cross obstacles etc..)
+    checkDestinationValidity(
+        map,
```

Comment:
Codacy found an issue: [Argument 'map' should be typed.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5941216)